optuna/samplers/_nsga2/crossover.py

```diff
@@ -0,0 +1,397 @@
+                child[param_name] = param
+                continue
+
+            trans = _SearchSpaceTransform({param_name: param_distribution})
+            transes.append(trans)
+            distributions.append(param_distribution)
+            param_names.append(param_name)
+            for parent_index, trial in enumerate(parents):
+                param = trans.transform({param_name: trial.params[param_name]})[0]
+                parents_not_categorical_params[parent_index].append(param)
+
+        xs = np.array(parents_not_categorical_params)
+
+        if crossover_name == "uniform":
+            params_array = _uniform(xs, rng, swapping_prob)
+        elif crossover_name == "blxalpha":
+            alpha = 0.5
+            params_array = _blxalpha(xs, rng, alpha)
+        elif crossover_name == "sbx":
+            if len(study.directions) == 1:
+                eta = 2
+            else:
+                eta = 20
+            params_array = _sbx(xs, rng, distributions, eta)
+        elif crossover_name == "vsbx":
+            if len(study.directions) == 1:
+                eta = 2
+            else:
+                eta = 20
+            params_array = _vsbx(xs, rng, eta)
+        elif crossover_name == "undx":
+            sigma_xi = 0.5
+            sigma_eta = 0.35 / np.sqrt(len(xs[0]))
+            params_array = _undx(xs, rng, sigma_xi, sigma_eta)
+        elif crossover_name == "undxm":
+            _m = 2
+            _n = len(xs[0])
+            sigma_xi = 1 / np.sqrt(_m)
+            sigma_eta = (
+                0.35
+                * np.sqrt(_m + 1)
+                * np.sqrt(3)
+                / np.sqrt(_n - _m)
+                / np.sqrt(_m + 2)
+                / np.sqrt(2)
+            )
+            params_array = _undxm(xs, rng, sigma_xi, sigma_eta)
+        elif crossover_name == "spx":
+            epsilon = np.sqrt(len(xs[0]) + 2)
+            params_array = _spx(xs, rng, epsilon)
+        else:
+            raise ValueError(f"{crossover_name} is not exist in optuna.")
+
+        _params = [
+            trans.untransform(np.array([param])) for trans, param in zip(transes, params_array)
+        ]
+        child = {}
+        for param in _params:
+            for param_name in param.keys():
+                child[param_name] = param[param_name]
+
+        if _check_in_constraints(child, search_space):
+            break
+
+    return child
+
+
+def _selection(
+    crossover_name: str,
+    study: Study,
+    parent_population: Sequence[FrozenTrial],
+    rng: np.random.RandomState,
+    dominates: Callable[[FrozenTrial, FrozenTrial, Sequence[StudyDirection]], bool],
+) -> List[FrozenTrial]:
+    if crossover_name in ["uniform", "blxalpha", "sbx", "vsbx"]:
+        n_select = 2
+    elif crossover_name in ["undx", "spx"]:
+        n_select = 3
+    elif crossover_name in ["undxm"]:
+        n_select = 4
+    else:
+        raise ValueError(f"{crossover_name} is not exist in optuna.")
```

Comment:
A nitpick. 
```suggestion
        raise ValueError(f"'{crossover_name}' is not a valid crossover name.")
```
It might be overkill but we could also include the list of acceptable crossover names in the message.

Reply:
I now noticed that we are performing a similar check in the sampler constructor. Can we simply omit raising an error here, maybe adding an assertion instead that the given crossover name is valid?